docs/TPM_with_LPC_protocol.md

```diff
@@ -0,0 +1,202 @@
+## TPM module with "Low Pin Count Protocol" build on SoC
+
+### Introduction
+
+The main goal of this document is showing the current state of development of TPM
+module with "LPC Protocol" based on SoC (System on Chip), and also documenting
+work carried out so far.
+
+## Motivation for doing this project
+
+We stiil see the necessity of making TPM module with "Low Pin Count" protocol.
+There is still many computers on the market (especcialy servers) which have
+mother-boards supporting LPC protocol.
+We considered possible hardware platforms in the past in this document:
+
+["Target TPM hardware interfaces"](https://lpntpm.lpnplant.io/hardware_interfaces/#low-pin-count-interface)
+
+Outside of fast "SPI Bus" and even "I2C" protocol which are relatively easy to
+implement on MCU haedware just "LPC protocol" make it difficult to implement on
+MCU.
+
+We made an attempt of implementation "LPC protocol" (LPC Peripheral) on MCU, but
+this attempt hasn't been successfully completed. "ESP32" board has been chosen as
+hardware platform. This hardware was Dual Core Tensilica LX6 with 240 MHz clock
+and 512 KB SRAM, which is quite powerful MCU. The maximum frequency of switching
+GPIO pins we could reach on this hardware was close to 1 MHz. The main reason for
+such bad result had been big latency in interrupts handling. We also considered
```

Comment:
We should link to all of the code we have developed in the process of testing that. We have used this repository, IIRC: https://github.com/lpn-plant/zephyr-lpc-app

We can link to the LPC implementations we have analyzed.

Lastly, we should upload some images, so this documentation is not boring. E.g. some photos of hardware, some screenshots of measured frequency of the signal, etc.